control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -3189,8 +3190,21 @@ func (r *HostedControlPlaneReconciler) reconcileOpenShiftAPIServer(ctx context.C
 		}
 	}
 
+	noProxy, err := proxy.DefaultNoProxy(hcp)
```

Comment:
I think DefaultNoProxy doesn't really need to return an err and we can save this check.